### ISSUE TITLE
unable to switch to npm package manger

### ISSUE THREAD

**Issue by code-brewer** [Microsoft/web-build-tools#1254]
I have a working rush demo,  then change to npm by edit **rush.json** :  comment out `"pnpmVersion": "2.15.1"`  and select  `"npmVersion": "4.5.0" `.

But comands like rush unlink, rush purge, rush update  give out Error:
`Rush Multi-Project Build Tool 5.7.1 - https://rushjs.io
ERROR: An unrecognized file "pnpmfile.js" was found in the Rush config folder: ~/rushDemo/common/config/rush`

If i delete common/config/rush/pnpmfile.js, then got error message :
`RROR: An unrecognized file "shrinkwrap.yaml" was found in the Rush config folder: ~/rushDemo/common/config/rush`

Can anyone show me how to use npm instead of pnpm ?

**Comment by octogonz** [comment]
Delete both those files.  (They are "unrecognized" because they're not used by NPM, and generally Rush doesn't want misleading files lying around to confuse people or tools.) 

Then run `rush update` to create an NPM-specific file.

Rush's behavior does seem a little misleading.  Perhaps we should document it better or improve the error message, if someone can propose a better behavior.

**Comment by code-brewer** [comment]
Thank for your help. It works.  Didn't expect i was just one step away from success,  but i was horrified and stop by the potentially endless files deletion. 

It would be better to note down this work-around in document even if can not gracefully solve this problem currently.

**Comment by nickpape** [comment]
Yeah, we could have Rush collect a list of all unrecognized files. Also, we store the last package manager that was used for an install in the `temp` folder, so we could detect the package manager change and leave a message like:

```
Rush has detected you may be switching from PNPM to NPM. In order to do so, Rush will need to delete the following PNPM-specific files:

-  /common/config/rush/pnpmfile.js
- /shrinkwrap.yaml
```

**Comment by gabrielliwerant** [comment]
Suggestion: Instead of `rush init` generating files based on pnpm by default, allow the user to select which package manager they use by prompt and then generate initial files accordingly. This would prevent confusion when switching a config value (which I would not expect to complain about auto-generated files) and would be in line with how `npm init` works.

**Comment by tmeindle** [comment]
upgrading pnpmVersion to 6.2.5 in rush.json also produces this message as well

**Comment by GongT** [comment]
pnpm 6 change `pnpmfile.js` to `.pnpmfile.cjs`, we need rename it.
